### DISCUSSION TITLE
Can't open file in wasm lib in wasi mode

### DISCUSSION THREAD

**Discussion by ActivePeter** [WasmEdge/WasmEdge#3117] [Q&A]
'called `Result::unwrap()` on an `Err` value: Custom { kind: Uncategorized, error: "failed to find a pre-opened file descriptor through which \"files/random_words.txt\" could be opened" }

files/random_words.txt is in the host running directory

vm create code:
``` rust
let config = ConfigBuilder::new(CommonConfigOptions::default())
    .with_host_registration_config(HostRegistrationConfigOptions::default().wasi(true))
    .build()
    .expect("failed to create config");
let module = Module::from_file(
    Some(&config),
    file_dir
        .as_ref()
        .join(format!("apps/{}/app.wasm", container_name)),
)
.unwrap();
// self.cache.lock().await.put(container_name, module.clone());
let import = host_funcs::new_import_obj();
let vm = VmBuilder::new()
    .with_config(config)
    .build()
    .expect("failed to create vm")
    .register_import_module(import)
    .unwrap();
let vm = vm
    .register_module(
        Some(&format!(
            "{}",
            self.next_instance_id.fetch_add(1, Ordering::Relaxed)
        )),
        module,
    )
    .unwrap();
```

**Comment by hydai** [comment]
Hi @apepkuss 
Could you please check this issue?

**Reply by ActivePeter** [reply]
I saw the process usage can use --dir to declare the position of root, but I didn't find related api in rust sdk.

**Comment by L-jasmine** [comment]
This is our new version of the[ Rust SDK](https://github.com/WasmEdge/wasmedge-rust-sdk). [Here is an example for WASI](https://github.com/second-state/wasmedge-rustsdk-examples/blob/refactor/new_sdk/async-wasi/run-wasm-app/src/main.rs)
